### PR TITLE
Show diversion diagrams in diversion alerts

### PR DESCRIPTION
**Pull request by anthonyshull**
https://app.asana.com/0/555089885850811/1207209404321117/f

### CODE REVIEW

File: test/cms/page/diversions_test.exs [hunk 1]
```diff
@@ -0,0 +1,50 @@
+defmodule CMS.Page.DiversionsTest do
```
Comment (anthonyshull) [inline test/cms/page/diversions_test.exs:1]: 📓 This is really just a copy of the `/page/basic_test.exs`. For tests related to the CMS, we have to use the static JSON responses until we rewrite all of the CMS tests.

File: test/cms/page_test.exs [hunk 2]
```diff
@@ -11,6 +11,12 @@ defmodule PageTest do
   describe "from_api/1" do
     test "switches on the node type in the json response and returns the proper page struct" do
       assert %Page.Basic{} = Page.from_api(Static.basic_page_response())
+
+      diversions_data =
+        Static.basic_page_response() |> Map.put("field_page_type", [%{"name" => "Diversions"}])
```
Comment (anthonyshull) [inline test/cms/page_test.exs:16]: 📓 This is just a hacky way of creating a Diversions page without creating a giant JSON blob like all other CMS test pages.

File: test/alerts/repo_test.exs [hunk 3]
```diff
@@ -54,6 +57,50 @@ defmodule Alerts.RepoTest do
...
+    end
+  end
+
+  describe "diversions_by_route_id/2" do
+    test "returns only diversions for the given route_ids" do
+      # Setup
+      diversion = Factories.Alerts.Alert.build(:alert, effect: :shuttle)
+      non_diversion = Factories.Alerts.Alert.build(:alert, effect: :delay)
+
+      Store.update([diversion, non_diversion], nil)
```
Comment (anthonyshull) [inline test/alerts/repo_test.exs:72]: 📓 We have to use this store until we convert the CMS code to using Mox for this.

**Review by amaisano — APPROVED** [review]
Relying on perfectly titled/named files is always risky, but it is an easy hack for now.

**Review by thecristen — APPROVED** [review]

File: lib/alerts/repo.ex [hunk 4]
```diff
@@ -55,4 +70,17 @@ defmodule Alerts.Repo do
     |> Store.all_alerts()
     |> Enum.filter(&(&1.priority == priority))
   end
+
+  # This function is used to attach an image URL to an alert if it doesn't already have one.
+  # It should be removed once alerts support image URLs.
+  defp maybe_attach_image_url(alert) do
+    if is_nil(alert.image_url) do
+      this_month = Timex.format!(Timex.now(), "%Y-%m", :strftime)
+      image_url = "/sites/default/files/media/#{this_month}/alert-#{alert.id}.png"
```
[the remark this answers is not in the document]
Comment (anthonyshull) [reply lib/alerts/repo.ex:79]: I expect we'll get a lot of cases of people asking us why the diagram isn't appearing.

**Pull request closed without merging**